tests/base_new_frmwrk.py

```diff
@@ -150,6 +150,217 @@ def set_replication_methods(self, conn_id, catalogs, replication_methods):
             connections.set_non_discoverable_metadata(
                 conn_id, catalog, menagerie.get_annotated_schema(conn_id, catalog['stream_id']), replication_md)
 
+    ### Method to return the fields that are non-upsert to be excluded from the all fields test as these non-upsert fields are not replicated
+    def get_upsert_only_fields(self, selected_fields, stream=None):
```

Comment:
How was this dict generated?  Are we sure it cannot be dynamic?  What if one of these fields changes?  Is there a way we can detect this programmatically?

Reply:
I generated this dict by looking at the failure and manually checking the tap-tester-target-out.json 
There is a method in runner.py - examine_target_output_for_fields which returns the fields with upsert message. 
See  line#132 of base_suite_tests/all_fields_test
This is the value we compare against. So, there is no point in calling the same method that.
Alternative is to override streams_to_selected_fields() 
We check this value and if not set, we get all fields
See: Line 99 of base_suite_tests/all_fields_test
Currently, what I am doing is the reverse of that.